### PR TITLE
PUBD-505 Add a test for broken images

### PR DESCRIPTION
**Pull request by hardyoyo**
Adds a test for broken images, closes PUBD-505

### CODE REVIEW

**Review by myucekul — COMMENTED** [review]
I'm concerned about Gemfile changes for this test. Can test be independent of Gemfile used for EB?

**Review by myucekul — COMMENTED** [review]
Can we also test for other assets links - like on submission page from slack messages about doj submissiongruidelines today.

File: .rubocop.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 
 AllCops:
-  TargetRubyVersion: 2.5
+  TargetRubyVersion: 3.0
```
Comment (tingletech) [inline .rubocop.yml:4]: why would this change for a test?

File: Gemfile.lock [hunk 2]
```diff
@@ -169,4 +184,4 @@ RUBY VERSION
    ruby 2.6.8p205
 
 BUNDLED WITH
-   2.2.27
+   2.3.8
```
Comment (tingletech) [inline Gemfile.lock:187]: why would this change for a test?

File: .rubocop.yml (hunk 1 above)
Comment (hardyoyo) [reply .rubocop.yml:4]: That's the version of Ruby we run on the beanstalks, and we run Rubocop along with our tests.

File: Gemfile.lock (hunk 2 above)
Comment (hardyoyo) [reply Gemfile.lock:187]: I'll drop this change, it's not necessary.

Comment (hardyoyo) [reply Gemfile.lock:187]: actually, it might be necessary, I had to add a few gems to facilitate the new test

Comment (hardyoyo) [reply Gemfile.lock:187]: I added at least one new gem to facilitate the image test, and ran bundle to install the gems. This is the result.

**Comment by hardyoyo** [comment]
> I'm concerned about Gemfile changes for this test. Can test be independent of Gemfile used for EB?

I had to add a couple gems to facilitate the new test for broken images. They're only used in the test, and should not impact the beanstalk at all.

**Comment by hardyoyo** [comment]
> Can we also test for other assets links - like on submission page from slack messages about doj submissiongruidelines today.

Yes, though the card PUB-505 does not currently mention such a requirement. My inclination would be to accept this test first, then improve it as part of a new card.

**Comment by hardyoyo** [comment]
> > Can we also test for other assets links - like on submission page from slack messages about doj submissiongruidelines today.
> 
> Yes, though the card PUBD-505 does not currently mention such a requirement. My inclination would be to accept this test first, then improve it as part of a new card.

Here's where we pull all the images out of the page we're testing:
https://github.com/eScholarship/jschol/pull/574/files#diff-8d97493c1f164c1a62f29f18a4e5b5ad7b4d6424854f8629d1d14f9c3296aaaaR174

We'd need to be able to search for the links inside the document, so we'd need to figure out the correct criteria for the search. That would be part of the future card we need to write up. We should probably also refactor this test to make the bulk of it into a function, so we can test multiple URLs for multiple kinds of things (images, other links, etc.).

**Review by tingletech — APPROVED** [review]

**Comment by hardyoyo** [comment]
Card [522](https://trello.com/c/r5xkFJtU) has been created to track the desired improvements mentioned above.

**Pull request merged**